package proto_ff;

import "yd_fieldoptions.proto";


message rolebornitemDesc
{
	optional int32 num = 1[(yd_fieldoptions.field_cname) = "数量"];
	optional int64 name = 2[(yd_fieldoptions.field_cname) = "名称"];
}

message rolebornmailDesc
{
	optional int32 num = 1[(yd_fieldoptions.field_cname) = "数量"];
	optional int64 name = 2[(yd_fieldoptions.field_cname) = "名称"];
}

message roleborn
{
	optional int64 bornID = 1[(yd_fieldoptions.field_cname) = "主键ID"];
	optional int32 professionID = 2[(yd_fieldoptions.field_cname) = "职业ID"];
	optional string professionName = 3[(yd_fieldoptions.field_cname) = "职业名称", (yd_fieldoptions.field_bufsize) = 60];
	optional string placement = 4[(yd_fieldoptions.field_cname) = "职业介绍", (yd_fieldoptions.field_bufsize) = 60];
	optional int32 gender = 5[(yd_fieldoptions.field_cname) = "性别"];
	optional int32 bornLevel = 6[(yd_fieldoptions.field_cname) = "出生等级"];
	optional int32 bornTitle = 7[(yd_fieldoptions.field_cname) = "携带称号"];
	optional int64 diamond = 8[(yd_fieldoptions.field_cname) = "携带钻石"];
	optional int64 bindDiamond = 9[(yd_fieldoptions.field_cname) = "携带绑定钻石"];
	optional int64 gold = 10[(yd_fieldoptions.field_cname) = "携带金币"];
	optional int64 mapID = 11[(yd_fieldoptions.field_cname) = "出生地图ID"];
	optional int32 birthFace = 12[(yd_fieldoptions.field_cname) = "出生朝向"];
	optional string icons = 13[(yd_fieldoptions.field_cname) = "头像图标", (yd_fieldoptions.field_bufsize) = 60];
	optional string loginAvata = 14[(yd_fieldoptions.field_cname) = "创角界面模型", (yd_fieldoptions.field_bufsize) = 60];
	optional string newweapon = 15[(yd_fieldoptions.field_cname) = "新手副本武器模型", (yd_fieldoptions.field_bufsize) = 60];
	optional string newclothing = 16[(yd_fieldoptions.field_cname) = "新手副本衣服模型", (yd_fieldoptions.field_bufsize) = 60];
	optional string weaponAvata = 17[(yd_fieldoptions.field_cname) = "出生时武器模型", (yd_fieldoptions.field_bufsize) = 60];
	optional string roleAvata = 18[(yd_fieldoptions.field_cname) = "出生时模型", (yd_fieldoptions.field_bufsize) = 60];
	optional int32 hairColor = 19[(yd_fieldoptions.field_cname) = "头发色值"];
	optional int32 clothesColor = 20[(yd_fieldoptions.field_cname) = "衣服色值"];
	optional int32 skinColor = 21[(yd_fieldoptions.field_cname) = "皮肤色值"];
	optional int64 carryWeapons = 22[(yd_fieldoptions.field_cname) = "装备武器"];
	optional int64 carryArmor = 23[(yd_fieldoptions.field_cname) = "装备衣服"];
	optional int64 carryCloak = 24[(yd_fieldoptions.field_cname) = "装备披风"];
	optional int64 carryShoes = 25[(yd_fieldoptions.field_cname) = "装备鞋子"];
	optional int64 carryNecklace = 26[(yd_fieldoptions.field_cname) = "装备项链"];
	optional int64 carryAmulet = 27[(yd_fieldoptions.field_cname) = "装备护符"];
	optional int64 carryBracer = 28[(yd_fieldoptions.field_cname) = "装备护腕"];
	optional int64 carryRing = 29[(yd_fieldoptions.field_cname) = "装备戒指"];
	optional int64 carryFashionWeapons = 30[(yd_fieldoptions.field_cname) = "装备时装（武器）"];
	optional int64 carryFashionArmor = 31[(yd_fieldoptions.field_cname) = "装备时装（衣服）"];
	optional int64 carryWings = 32[(yd_fieldoptions.field_cname) = "翅膀"];
	optional int32 mSpeed = 33[(yd_fieldoptions.field_cname) = "初始移动速度"];
	optional int32 view = 34[(yd_fieldoptions.field_cname) = "视野"];
	optional int32 radius = 35[(yd_fieldoptions.field_cname) = "人物半径"];
	optional int32 group = 36[(yd_fieldoptions.field_cname) = "技能组"];
	optional int64 sheny = 37[(yd_fieldoptions.field_cname) = "神装衣服"];
	optional int64 shenw = 38[(yd_fieldoptions.field_cname) = "神装武器"];
	repeated int32 actBuff = 39[(yd_fieldoptions.field_cname) = "生成buff", (yd_fieldoptions.field_arysize) = 4];
	repeated rolebornitemDesc item = 40[(yd_fieldoptions.field_cname) = "背包物品", (yd_fieldoptions.field_arysize) = 35];
	repeated int64 actSkill = 41[(yd_fieldoptions.field_cname) = "生成技能", (yd_fieldoptions.field_arysize) = 12];
	repeated rolebornmailDesc mail = 42[(yd_fieldoptions.field_cname) = "邮件物品", (yd_fieldoptions.field_arysize) = 1];
}


message Sheet_roleborn
{
	repeated roleborn roleborn_List = 1[(yd_fieldoptions.field_arysize)=20];
}

message rolemasterMaleattributeDesc
{
	optional int32 Type = 1[(yd_fieldoptions.field_cname) = "类型"];
	optional int32 Value = 2[(yd_fieldoptions.field_cname) = "值"];
}

message rolemasterMale
{
	optional int32 lv = 1[(yd_fieldoptions.field_cname) = "玩家等级"];
	optional int32 upgradePoint = 2[(yd_fieldoptions.field_cname) = "升级获得点数"];
	repeated rolemasterMaleattributeDesc attribute = 3[(yd_fieldoptions.field_cname) = "属性", (yd_fieldoptions.field_arysize) = 70];
}


message Sheet_rolemasterMale
{
	repeated rolemasterMale rolemasterMale_List = 1[(yd_fieldoptions.field_arysize)=2000];
}

message rolemasterFemaleattributeDesc
{
	optional int32 Type = 1[(yd_fieldoptions.field_cname) = "类型"];
	optional int32 Value = 2[(yd_fieldoptions.field_cname) = "值"];
}

message rolemasterFemale
{
	optional int32 lv = 1[(yd_fieldoptions.field_cname) = "玩家等级"];
	optional int32 upgradePoint = 2[(yd_fieldoptions.field_cname) = "升级获得点数"];
	repeated rolemasterFemaleattributeDesc attribute = 3[(yd_fieldoptions.field_cname) = "属性", (yd_fieldoptions.field_arysize) = 22];
}


message Sheet_rolemasterFemale
{
	repeated rolemasterFemale rolemasterFemale_List = 1[(yd_fieldoptions.field_arysize)=2000];
}

message rolesickleMaleattributeDesc
{
	optional int32 Type = 1[(yd_fieldoptions.field_cname) = "类型"];
	optional int32 Value = 2[(yd_fieldoptions.field_cname) = "值"];
}

message rolesickleMale
{
	optional int32 lv = 1[(yd_fieldoptions.field_cname) = "玩家等级"];
	optional int32 upgradePoint = 2[(yd_fieldoptions.field_cname) = "升级获得点数"];
	repeated rolesickleMaleattributeDesc attribute = 3[(yd_fieldoptions.field_cname) = "属性", (yd_fieldoptions.field_arysize) = 22];
}


message Sheet_rolesickleMale
{
	repeated rolesickleMale rolesickleMale_List = 1[(yd_fieldoptions.field_arysize)=2000];
}

message rolesickleFemaleattributeDesc
{
	optional int32 Type = 1[(yd_fieldoptions.field_cname) = "类型"];
	optional int32 Value = 2[(yd_fieldoptions.field_cname) = "值"];
}

message rolesickleFemale
{
	optional int32 lv = 1[(yd_fieldoptions.field_cname) = "玩家等级"];
	optional int32 upgradePoint = 2[(yd_fieldoptions.field_cname) = "升级获得点数"];
	repeated rolesickleFemaleattributeDesc attribute = 3[(yd_fieldoptions.field_cname) = "属性", (yd_fieldoptions.field_arysize) = 22];
}


message Sheet_rolesickleFemale
{
	repeated rolesickleFemale rolesickleFemale_List = 1[(yd_fieldoptions.field_arysize)=2000];
}

message roleswordMaleattributeDesc
{
	optional int32 Type = 1[(yd_fieldoptions.field_cname) = "类型"];
	optional int32 Value = 2[(yd_fieldoptions.field_cname) = "值"];
}

message roleswordMale
{
	optional int32 lv = 1[(yd_fieldoptions.field_cname) = "玩家等级"];
	optional int32 upgradePoint = 2[(yd_fieldoptions.field_cname) = "升级获得点数"];
	repeated roleswordMaleattributeDesc attribute = 3[(yd_fieldoptions.field_cname) = "属性", (yd_fieldoptions.field_arysize) = 22];
}


message Sheet_roleswordMale
{
	repeated roleswordMale roleswordMale_List = 1[(yd_fieldoptions.field_arysize)=2000];
}

message roleswordFemaleattributeDesc
{
	optional int32 Type = 1[(yd_fieldoptions.field_cname) = "类型"];
	optional int32 Value = 2[(yd_fieldoptions.field_cname) = "值"];
}

message roleswordFemale
{
	optional int32 lv = 1[(yd_fieldoptions.field_cname) = "玩家等级"];
	optional int32 upgradePoint = 2[(yd_fieldoptions.field_cname) = "升级获得点数"];
	repeated roleswordFemaleattributeDesc attribute = 3[(yd_fieldoptions.field_cname) = "属性", (yd_fieldoptions.field_arysize) = 22];
}


message Sheet_roleswordFemale
{
	repeated roleswordFemale roleswordFemale_List = 1[(yd_fieldoptions.field_arysize)=2000];
}

message roletaidaoMaleattributeDesc
{
	optional int32 Type = 1[(yd_fieldoptions.field_cname) = "类型"];
	optional int32 Value = 2[(yd_fieldoptions.field_cname) = "值"];
}

message roletaidaoMale
{
	optional int32 lv = 1[(yd_fieldoptions.field_cname) = "玩家等级"];
	optional int32 upgradePoint = 2[(yd_fieldoptions.field_cname) = "升级获得点数"];
	repeated roletaidaoMaleattributeDesc attribute = 3[(yd_fieldoptions.field_cname) = "属性", (yd_fieldoptions.field_arysize) = 22];
}


message Sheet_roletaidaoMale
{
	repeated roletaidaoMale roletaidaoMale_List = 1[(yd_fieldoptions.field_arysize)=2000];
}

message roletaidaoFemaleattributeDesc
{
	optional int32 Type = 1[(yd_fieldoptions.field_cname) = "类型"];
	optional int32 Value = 2[(yd_fieldoptions.field_cname) = "值"];
}

message roletaidaoFemale
{
	optional int32 lv = 1[(yd_fieldoptions.field_cname) = "玩家等级"];
	optional int32 upgradePoint = 2[(yd_fieldoptions.field_cname) = "升级获得点数"];
	repeated roletaidaoFemaleattributeDesc attribute = 3[(yd_fieldoptions.field_cname) = "属性", (yd_fieldoptions.field_arysize) = 22];
}


message Sheet_roletaidaoFemale
{
	repeated roletaidaoFemale roletaidaoFemale_List = 1[(yd_fieldoptions.field_arysize)=2000];
}

message roleexp
{
	optional int32 lv = 1[(yd_fieldoptions.field_cname) = "等级"];
	optional int64 exp = 2[(yd_fieldoptions.field_cname) = "当前等级升级需要经验"];
	optional int32 skillPoint = 3[(yd_fieldoptions.field_cname) = "天赋点"];
}


message Sheet_roleexp
{
	repeated roleexp roleexp_List = 1[(yd_fieldoptions.field_arysize)=2000];
}

message roleskillType
{
	optional int64 skillId = 1[(yd_fieldoptions.field_cname) = "技能ID"];
	optional int32 skillType = 2[(yd_fieldoptions.field_cname) = "技能类型"];
}


message Sheet_roleskillType
{
	repeated roleskillType roleskillType_List = 1[(yd_fieldoptions.field_arysize)=200];
}

message roleaddSkill
{
	optional int32 Id = 1[(yd_fieldoptions.field_cname) = "ID"];
	optional int64 skillId = 2[(yd_fieldoptions.field_cname) = "技能类型"];
}


message Sheet_roleaddSkill
{
	repeated roleaddSkill roleaddSkill_List = 1[(yd_fieldoptions.field_arysize)=20];
}

message roleworldexp
{
	optional int64 lv = 1[(yd_fieldoptions.field_cname) = "等级差"];
	optional int32 worldExp = 2[(yd_fieldoptions.field_cname) = "经验加成"];
}


message Sheet_roleworldexp
{
	repeated roleworldexp roleworldexp_List = 1[(yd_fieldoptions.field_arysize)=600];
}
